src/cdk/stepper/stepper.ts

```diff
@@ -150,7 +152,7 @@ export class CdkStepper {
   @Input()
   get selected() { return this._steps[this.selectedIndex]; }
   set selected(step: CdkStep) {
-    let index = this._steps.toArray().indexOf(step);
+    let index = this._stepsArray.indexOf(step);
```

Comment:
You can inline this assignment: `this.selectedIndex = this._stepsArray.indexOf(step)`.